src/main/java/com/adobe/target/edge/client/service/TelemetryService.java

```diff
@@ -51,6 +54,24 @@ public void addTelemetry(
     }
   }
 
+  public void addTelemetry(
+      TargetDeliveryRequest deliveryRequest,
+      TimingTool timer,
+      TargetDeliveryResponse targetDeliveryResponse,
+      double parsingTime,
+      long responseSize) {
+    TelemetryEntry telemetryEntry =
+        createTelemetryEntry(
+            deliveryRequest, targetDeliveryResponse, timer.timeEnd(TIMING_EXECUTE_REQUEST));
+    if (telemetryEntry != null) {
```

Comment:
Let's invert the condition here
```suggestion
    if (telemetryEntry == null) {
        return;
    }
```